hydra/core/override_parser/overrides_visitor.py

```diff
@@ -173,6 +184,8 @@ def visitDictKeyValuePair(
     ) -> Tuple[str, ParsedElementType]:
         children = ctx.getChildren()
         item = next(children)
+        # We should never visit dictionaries that have interpolations as keys
+        # (currently these are only allowed within interpolations, which are not visited).
```

Comment:
I don't understand this comment.
1. Since when do we have interpolation as keys in dictionaries?
2. is the code doing anything to address the requirements from this comment?

Reply:
I'll try to make this comment clearer if I can convince you to keep interpolations as keys ;) But to answer your questions:

1. Since https://github.com/omry/omegaconf/pull/321
2. That's actually the assert just below this comment (but I'm not entirely sure I understood your question)

For additional context:
- In OmegaConf, when writing the grammar I added support for dictionaries with interpolations as keys. Such dictionaries can only happen when inside an interpolation.
- When "merging" OmegaConf's grammar within Hydra, I wondered what to do if someone wanted to define a dictionary with interpolations as keys *outside* of an interpolation. To be honest I didn't spend much time thinking about it and decided just not to support it for now out of simplicity (thus keeping the assert below the comment).

Reply:
1. Since I didn't really get to review it yet, I was not aware of this intention. can you show me a convincing use case for it?
2. The assert is the answer to my second question. the comment like an unrelated piece of information. I did not realize the assert was really about enforcing it.